owncloudApp/src/main/java/com/owncloud/android/presentation/ui/authentication/LoginActivity.kt

```diff
@@ -347,26 +367,79 @@ class LoginActivity : AppCompatActivity(), SslUntrustedCertDialog.OnSslUntrusted
 
     /**
      * OAuth step 1: Get authorization code
+     * Firstly, try the OAuth authorization with Open Id Connect, checking whether there's an available .well-known url
+     * to use or not
      */
-    private fun startOAuthorization() {
+    private fun startOIDCOauthorization() {
         server_status_text.run {
             setCompoundDrawablesWithIntrinsicBounds(R.drawable.progress_small, 0, 0, 0)
             text = resources.getString(R.string.oauth_login_connection)
         }
 
-        mAuthService = OAuthUtils.createAuthorizationService(this)
+        val retrieveConfigurationCallback =
+            RetrieveConfigurationCallback { serviceConfiguration, exception ->
+                if (exception != null) {
+                    Timber.e(exception, "OIDC failed, try with normal OAuth");
+                    startNormalOauthorization();
+                } else if (serviceConfiguration != null) {
+                    oidcSupported = true;
+                    performGetAuthorizationCodeRequest(serviceConfiguration)
+                    authorizationServiceConfiguration = serviceConfiguration
+                }
+            }
+
+        OAuthUtils.buildOIDCAuthorizationServiceConfig(
+            this,
+            serverBaseUrl,
+            retrieveConfigurationCallback
+        )
+    }
+
+    /**
+     * OAuth step 1: Get authorization code
+     * If OIDC is not available, falling back to normal OAuth
+     */
+    private fun startNormalOauthorization() {
```

Comment:
This method is pretty similar to `startOIDCOauthorization` can't they merge into one?

BTW, i cannot see here a `server_status_text` update, user can't see what's happening, can he?

Reply:
> This method is pretty similar to startOIDCOauthorization

Not really, take a deeper look, the lines that are exactly the same, are just these three:

oidcSupported = true
performGetAuthorizationCodeRequest(serviceConfiguration)
authorizationServiceConfiguration = serviceConfiguration

And I've just realized `oidcSupported = true` should not be included in the normal OAuth case, copying-pasting issue

Other checks or call methods seem the same but are different either on the params used or the call itself.

> BTW, i cannot see here a server_status_text update, user can't see what's happening, can he?

We always try the oidc connection before calling this method, so the status update takes place in the startOIDCOauthorization method.